### PR TITLE
mariadb: 10.3.20 -> 10.3.22

### PR DESCRIPTION
**Pull request by Izorkin**
###### Motivation for this change
Update package to version 10.3.22
Fixed this warnings:
```-- WITH_NUMA=OFF: NUMA memory allocation policy disabled```
and
```
-- Checking for module 'kytea'
--   No package 'kytea' found
-- Looking for libstemmer.h
-- Looking for libstemmer.h - not found
-- Looking for sb_stemmer_list in stemmer
-- Looking for sb_stemmer_list in stemmer - not found
-- Checking for module 'libzmq'
--   No package 'libzmq' found
-- Looking for event_init in event
-- Looking for event_init in event - found
-- Checking for module 'msgpack'
--   No package 'msgpack' found
-- Looking for msgpack_version in msgpack
-- Looking for msgpack_version in msgpack - not found
```
Add custom options - `withNUMA` and `withStorageMroonga `
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Izorkin** [comment]
сс @aanderse @Mic92

File: pkgs/servers/sql/mariadb/default.nix [hunk 1]
```diff
@@ -1,7 +1,9 @@
 { stdenv, fetchurl, fetchFromGitHub, cmake, pkgconfig, makeWrapper, ncurses, zlib, xz, lzo, lz4, bzip2, snappy
 , libiconv, openssl, pcre, boost, judy, bison, libxml2, libkrb5, linux-pam, curl
-, libaio, libevent, jemalloc, cracklib, systemd, numactl, perl
+, libaio, libevent, jemalloc, cracklib, systemd, perl
 , fixDarwinDylibNames, cctools, CoreServices, less
+, withNUMA ? true, numactl
```
Comment (Mic92) [inline pkgs/servers/sql/mariadb/default.nix]: I would say that numa is common enough in the area where database servers are used that we don't even need to add an option for it. Just drop the `withNUMA` option until someone finds a good reason for not having it.

Comment (Izorkin) [reply pkgs/servers/sql/mariadb/default.nix]: Fixed

**Comment by aanderse** [comment]
This release fixes [CVE-2020-2574](http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-2574) and so the version bump commit should probably be back ported to `20.03`.

**Comment by Mic92** [comment]
20.03: 153baa2674010416bed78f797f8239e1cbfed956

**Comment by aanderse** [comment]
Thanks @Izorkin and @Mic92!

**Pull request merged**